src/shared/Core/Settings.cs

```diff
@@ -559,6 +564,22 @@ public bool IsInteractionAllowed
                 KnownGitCfg.Credential.Trace,
                 out value) && !value.IsFalsey();
 
+        public bool UseSoftwareRendering
+        {
+            get
+            {
+                // WORKAROUND: Some Windows ARM devices have a graphics driver issue that causes transparent windows
+                // when using hardware rendering. Until this is fixed, we will default to software rendering on these
+                // devices. Users can always override this setting back to HW-accelerated rendering if they wish.
+                bool defaultValue = PlatformUtils.IsWindows() && PlatformUtils.IsArm();
+
+                return TryGetSetting(KnownEnvars.GcmGuiSoftwareRendering,
+                    KnownGitCfg.Credential.SectionName,
+                    KnownGitCfg.Credential.GuiSoftwareRendering,
+                        out string str) && str.ToBooleanyOrDefault(defaultValue);
```

Comment:
Does this `&&` mean this will be restricted only to devices where the `defaultValue` is `true` (i.e. Windows ARM devices)?

Reply:
This Code is certainly not ideal. It seems like `defaultValue` is intended to be used when no value (or a bogus value) is explicitly configured. But when no value is explicitly configured, `TryGetSetting()` will return `false` and `str` will be `null`, so the `&&` will short circuit and return `false`, despite `defaultValue` potentially being `true`.

Reply:
> It seems like defaultValue is intended to be used when no value (or a bogus value) is explicitly configured.

Correct. This was the _intent_.. but also correct that this isn't what it's doing! 🤦‍♂️ 

Updated the logic to return the default value either when there's no setting (`TryGet` returns false) or a bad value (`ToBooleany` returns the default).